aspnetcore/blazor/tutorials/movie-database-app/part-8.md

```diff
@@ -0,0 +1,272 @@
+- <PropertyColumn Property="movie => movie.Title" />
++ <PropertyColumn Property="movie => movie.Title" Sortable="true" />
+```
+
+You can sort the `QuickGrid` by movie title by selecting the **:::no-loc text="Title":::** column. The page doesn't reload for sorting to occur. The sorting is performed live over the SignalR connection, where the sorting operation is performed on the server with the rendered result sent back to the client:
+
+![Movie list sorted by the Title column](~/blazor/tutorials/movie-database-app/part-8/_static/sorted-movies.png)
+
+## Use C# code and interactivity to search by title
+
+In an earlier part of the tutorial series, the `Index` component was modified to allow the user to filter movies by title. This was accomplished by:
+
+* Adding an HTML form that issues a GET request to the server with the user's title search string as a query string field-value pair (for example, `?titleFilter=road+warrior` if the user searches for "`road warrior`"):
+
+  ```html
+  <form action="/movies">
+      <input type="search" name="titleFilter" />
+      <input type="submit" value="Search" />
+  </form>
+  ```
+
+* Adding code to the component that obtains the title search string from the query string and uses it to filter the database's records:
+
+  ```csharp
+  [SupplyParameterFromQuery]
+  public string? TitleFilter { get; set; }
+
+  protected override void OnParametersSet()
+  {
+      if (!string.IsNullOrEmpty(TitleFilter))
+      {
+          movies = DB.Movie.Where(
+              s => !string.IsNullOrEmpty(s.Title) ? 
+                  s.Title.Contains(TitleFilter) : false);
+      }
+      else
+      {
+          movies = DB.Movie;
+      }
+  }
+  ```
+
+The preceding approach is effective for a component that adopts static SSR, where the only interaction between the client and server is via HTTP requests. There was no live SignalR connection between the client and the server, and there was no way for the app on the server to process C# code *interactively* based on the user's actions in the component's UI and return content without a full-page reload.
+
+Now that the component is interactive, it can provide an improved user experience with Blazor features for binding and event handling, where full-page reloads aren't required to run C# on the server that interacts with the page's elements.
+
+Add a delegate event handler that the user can trigger to filter the database's movie records. The method uses the value of the `TitleFilter` property to perform the operation. If the user clears `TitleFilter` and searches, the method loads the entire movie list for display.
+
+Add the following method to the `@code` block:
+
+```csharp
+private void FilterMovies()
+{
+    if (!string.IsNullOrEmpty(TitleFilter))
+    {
+        movies = DB.Movie.Where(
+            s => !string.IsNullOrEmpty(s.Title) ? 
+                s.Title.Contains(TitleFilter) : false);
+    }
+    else
+    {
+        movies = DB.Movie;
+    }
+}
+```
```

Comment:
I think this code can be simplified. In the past I've demoed search functionality like this using a helper property like this:

```csharp
string titleFilter = "";
IQueryable<Movie> FilteredMovies => DB.Movie.Where(m => m.Title.Contains(titleFilter));
```

This works because `Contains` returns true when you pass it an empty string.

Reply:
If you go with the code I suggested in the earlier section on setting up search filtering, then I think all you need to do here is remove the `[SupplyParameterFromQuery]` attribute and instead bind the `TitleFilter` to the input.

Reply:
I especially ❤️ this approach with `@bind:event="oninput"` for per-character live search filtering. I'll refactor for this approach.